CONTRIBUTING.md

```diff
@@ -1,6 +1,38 @@
 
 # Contributing guidelines
 
+## Docker for refinebio-examples
+
+This repository uses a Docker image to manage dependencies and keep software versions consistent.
+Development should take place in the Docker container.
+
+### Setting up the docker container
+
+You can build the docker image locally using:
+```
+docker build -< Dockerfile -t refinebio-examples
+```
+Or by pulling the image from Docker hub.
+```
+docker pull ccdl/refinebio-examples
+```
+Run the container with whatever method you choose.
+
+**With Kitematic**
+```
+docker run -e PASSWORD=<PASSWORD> -p 8787:8787 refinebio-examples
+```
+Then navigate to Kitematic to set your volume.
+Then you can navigate to http://localhost:8787/ to start developing.
+
+**Without Kitematic**
+
+Replace `<LOCAL_PATH_TO_REFINEBIO-EXAMPLES_REPO>` and run.
+```
+docker run -it --rm --mount type=volume,dst=/home/rstudio/kitematic,volume-driver=local,volume-opt=type=none,volume-opt=o=bind,volume-opt=device=<LOCAL_PATH_TO_REFINEBIO-EXAMPLES_REPO> -e PASSWORD=<PASSWORD> -p 8787:8787 refinebio-examples
```

Comment:
I think we can/should simplify this. Notice that I am not using `/kitematic`, because I don't think it is needed, and just adds a level of confusion since you have to change directories to work with your local files. This way RStudio opens directly to the path you give. 
```suggestion
docker run --mount type=bind,target=/home/rstudio,source=<LOCAL_PATH_TO_REFINEBIO-EXAMPLES_REPO> -e PASSWORD=<PASSWORD> -p 8787:8787 refinebio-examples
```
We also should not need `-it` unless we are running a separate command and need to interact with the terminal, which we are not for general use. We also don't generally want to remove the image after we quit, so no `--rm`.

For the `snakemake` instructions, you could potentially add the following which will run `snakemake` in the correct directory without needing a web session. (`-it` could be used here, but should not be necessary)
```
docker run -w /home/rstudio --mount type=bind,target=/home/rstudio,source=<LOCAL_PATH_TO_REFINEBIO-EXAMPLES_REPO> -e  refinebio-examples snakemake
```

Or, if you make the WORKDIR suggestion I added to the Dockerfile:

```
docker run --mount type=bind,target=/home/rstudio,source=<LOCAL_PATH_TO_REFINEBIO-EXAMPLES_REPO> -e  refinebio-examples snakemake
```

Reply:
Even easier, if you are in the refinebio-examples directory: 
```
docker run --mount type=bind,target=/home/rstudio,source=$PWD refinebio-examples snakemake
```

Reply:
Awesome! I knew there was a way to simplify this but wasn't sure how.

Reply:
I realized this may actually need to be 
```
docker run --mount type=bind,target=/home/rstudio,source=$PWD refinebio-examples snakemake --cores 1
```
If we use current `snakemake` & other python packages. (which we should)